### PR TITLE
Update analytics card header text styles

### PR DESCRIPTION
**Pull request by LevinMedia**
This PR updates the analytics card header text to use the `subtitle` styles from the @wordpress/components package. Also minor updates were made to the chart header titles for size color and margin. 

### Accessibility

-   [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots

**Before**

Screen Shot 2021-03-03 at 12 55 35 PM

Screen Shot 2021-03-03 at 12 55 11 PM

Screen Shot 2021-03-03 at 12 55 22 PM

**After**

Screen Shot 2021-03-03 at 12 54 18 PM

Screen Shot 2021-03-03 at 12 54 36 PM

Screen Shot 2021-03-03 at 12 53 04 PM

### Detailed test instructions:

-   Run this branch
-   Verify that the analytics card header sizes have been updated, and that the color and sizing are consistent across the experience.

### CODE REVIEW

File: packages/components/src/chart/style.scss [hunk 1]
```diff
@@ -24,11 +24,11 @@
 
 		.woocommerce-chart__title {
 			height: 18px;
-			color: $wp-admin-sidebar;
-			font-size: 15px;
+			color: #23282d;
```
Comment (becdetat) [inline packages/components/src/chart/style.scss:27]: Is there no color variable for this?

**Review by becdetat — COMMENTED** [review]
Everything looks good, both code and app. I just had the one comment about hard-coding the colour and there is a failing test here: https://github.com/woocommerce/woocommerce-admin/pull/6506/checks?check_run_id=2026007823#step:5:3850

The test seems to be related to changing the class of the `h2` caused by changing the `Text` variant.

**Comment by moon0326** [comment]
cc @LevinMedia as this PR has been idle for more than 26 days.

**Review by becdetat — APPROVED** [review]
Per DM with @LevinMedia I'll approve this, the colour value isn't a deal breaker and we don't import that colour anywhere else in WCA.

We also don't need the testing instructions in `TESTING-INSTRUCTIONS.md`. 

Thanks :-D

**Comment by joshuatf** [comment]
Hey @LevinMedia 👋  Do you think you'll have time to work on this or do you need someone to take over?  In addition to @becdetat's comment, this is going to need a rebase to fix conflicts.

**Comment by joshuatf** [comment]
Re-pinging @LevinMedia - we have cooldown coming up so if you don't have the time, I think someone from Mothra can take over.

**Comment by joshuatf** [comment]
@becdetat I rebased this PR and made some updates given how Gutenberg handles "variants" now.  Could you give this another look?

The only place I could find the color variable used is in the dark grays in GB, but they are not exported as far as I can tell.  The new `Text` component updates seem to favor css-in-js, but we don't yet have any color vars in JS.  Open to feedback on this if we want to begin adding them.

**Review by becdetat — APPROVED** [review]
@joshuatf Everything looks good 🚢

**Pull request merged**